### PR TITLE
Fixing Issue #340

### LINKED ISSUE #340: [BUG] Warnings C4715 in CurveConverter
**Describe the bug**
Not a stopper, but the recent changes in CurveConverter now provide C4715 warnings.

> curveconverter.h(1883): warning C4715: ... not all control paths return a value

This should be addressed.

### PR DESCRIPTION
**Pull request by SamuilsRulovs**
Fixes #340 

Correcting functions  getAngleOnEllipse, getAngleOnCircle and convertIfcArcIndex, so these functions do not provide C4715 warnings. 

Refactored functions  getAngleOnEllipse, getAngleOnCircle and convertIfcArcIndex using function areEqual

### CODE REVIEW

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 1]
```diff
@@ -678,8 +678,7 @@ namespace OpenInfraPlatform {
 					double xDeltaA = arcMid2D.x - arcStart2D.x;
 					double yDeltaB = arcEnd2D.y - arcMid2D.y;
 					double xDeltaB = arcEnd2D.x - arcMid2D.x;
-
-					if (xDeltaA != 0. && xDeltaB != 0.) {
+					if (!GeomSettings()->areEqual(xDeltaA, 0.) && !GeomSettings()->areEqual(xDeltaB, 0.)){
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h]: Here it is unnecessary to check if the values are "close to" zero - they really just must not be zero. So while such change is welcomed, here it is out of order. Please return to the original.

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 2]
```diff
@@ -1848,17 +1850,21 @@ namespace OpenInfraPlatform {
 				) const throw(...)
 				{
 					carve::geom::vector<3> centerToTrimPoint = trimPoint - circleCenter;
-					if (abs(centerToTrimPoint.length() - circleRadius) < 0.0001) {
+
+					if (GeomSettings()->areEqual(abs(centerToTrimPoint.length()), circleRadius)){
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h]: `.length()` is always positive, no need for `abs(..)`.

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 3]
```diff
@@ -1848,17 +1850,21 @@ namespace OpenInfraPlatform {
...
 				{
 					carve::geom::vector<3> centerToTrimPoint = trimPoint - circleCenter;
-					if (abs(centerToTrimPoint.length() - circleRadius) < 0.0001) {
+
+					if (GeomSettings()->areEqual(abs(centerToTrimPoint.length()), circleRadius)){
 						centerToTrimPoint.normalize();
 						double cosAngle = carve::geom::dot(centerToTrimPoint, carve::geom::vector<3>(carve::geom::VECTOR(1., 0., 0.)));
 
-						if (abs(cosAngle) < 0.0001) {
+						if (GeomSettings()->areEqual(abs(cosAngle), 0.)) {
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h]: Always call `this->GeomSetting()...` - it is an error with vs2019, which we will be switching to shortly.

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 4]
```diff
@@ -1901,13 +1907,16 @@ namespace OpenInfraPlatform {
...
 							if (centerToTrimPoint.y > 0.) {
 								return M_PI_2;
 							}
 							else if (centerToTrimPoint.y < 0.) {
 								return 3 * M_PI_2;
 							}
+							else {
+								throw oip::InconsistentGeometryException("Cosinus and sinus cannot be 0 simultaneously!");
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h]: In English it is "cosine" and "sine" :)

File: Core/src/IfcGeometryConverter/CurveConverter.h [hunk 5]
```diff
@@ -1901,13 +1907,16 @@ namespace OpenInfraPlatform {
 					if ((centerToTrimPoint.x / ellipseRadiusX) <= 1. && (centerToTrimPoint.y / ellipseRadiusY) <= 1.) {
 						double cosAngle = centerToTrimPoint.x / ellipseRadiusX;
 
-						if (abs(cosAngle) < 0.0001) {
+						if (GeomSettings()->areEqual(abs(cosAngle), 0.)) {
 							if (centerToTrimPoint.y > 0.) {
 								return M_PI_2;
 							}
 							else if (centerToTrimPoint.y < 0.) {
 								return 3 * M_PI_2;
```
Comment (pjanck) [inline Core/src/IfcGeometryConverter/CurveConverter.h]: multiplying integer with double?

**Review by pjanck — CHANGES_REQUESTED** [review]

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 1 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/CurveConverter.h]: Resolved: 6e54eae

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 2 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/CurveConverter.h]: Resolved: 6e54eae

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 3 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/CurveConverter.h]: Resolved: 6e54eae

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 4 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/CurveConverter.h]: Resolved: 6e54eae

File: Core/src/IfcGeometryConverter/CurveConverter.h (hunk 5 above)
Comment (SamuilsRulovs) [reply Core/src/IfcGeometryConverter/CurveConverter.h]: Resolved: 6e54eae

**Review by pjanck — APPROVED** [review]

**Pull request merged**